src/internal/common/dttm/dttm_test.go

```diff
@@ -58,6 +58,49 @@ func (suite *DTTMUnitSuite) TestParseTime() {
 	require.Error(t, err, clues.ToCore(err))
 }
 
+func (suite *DTTMUnitSuite) TestOrNow() {
+	var (
+		oneMinuteAgo = time.Now().Add(-1 * time.Minute)
+		inOneMinute  = time.Now().Add(1 * time.Minute)
+		inOneHour    = time.Now().Add(1 * time.Hour)
+		empty        = time.Time{}
+	)
+
+	table := []struct {
+		name   string
+		input  time.Time
+		expect func(t *testing.T, result time.Time)
+	}{
+		{
+			name:  "a minute ago",
+			input: oneMinuteAgo,
+			expect: func(t *testing.T, result time.Time) {
+				assert.True(t, oneMinuteAgo.Equal(result), "equal")
+			},
+		},
+		{
+			name:  "in an hour",
+			input: inOneHour,
+			expect: func(t *testing.T, result time.Time) {
+				assert.True(t, inOneHour.Equal(result), "equal")
+			},
+		},
+		{
+			name:  "empty",
+			input: empty,
+			expect: func(t *testing.T, result time.Time) {
+				assert.True(t, oneMinuteAgo.Before(result), "lower bound")
+				assert.True(t, inOneMinute.After(result), "upper bound")
```

Comment:
would you be interested in the `WithinDuration` [assertion](https://pkg.go.dev/github.com/stretchr/testify/assert#WithinDuration)?

Reply:
hey, that's nifty!

Reply:
Not a bad idea.